enclave/src/rpc/rpc_place_order.rs

```diff
@@ -17,43 +17,65 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 pub extern crate alloc;
-use alloc::{string::String, string::ToString};
+use alloc::{boxed::Box, string::String, string::ToString};
 
+use crate::rpc::polkadex_rpc_gateway::RpcGateway;
 use crate::rpc::rpc_call_encoder::{JsonRpcCallEncoder, RpcCall, RpcCallEncoder};
-use crate::rpc::rpc_info::{RpcCallStatus, RpcInfo};
-use crate::rpc::trusted_operation_verifier::get_verified_trusted_operation;
+use crate::rpc::rpc_info::RpcCallStatus;
+use crate::rpc::trusted_operation_verifier::TrustedOperationExtractor;
 use jsonrpc_core::{BoxFuture, Params, Result as RpcResult, RpcMethodSync, Value};
 use log::*;
-use polkadex_sgx_primitives::types::DirectRequest;
+use polkadex_sgx_primitives::types::{DirectRequest, OrderUUID};
 use substratee_stf::{TrustedCall, TrustedOperation};
 use substratee_worker_primitives::DirectRequestStatus;
 
-pub struct RpcPlaceOrder {}
+pub struct RpcPlaceOrder {
+    top_extractor: Box<dyn TrustedOperationExtractor + 'static>,
+    rpc_gateway: Box<dyn RpcGateway + 'static>,
+}
 
 impl RpcPlaceOrder {
+    pub fn new(
+        top_extractor: Box<dyn TrustedOperationExtractor + 'static>,
+        rpc_gateway: Box<dyn RpcGateway + 'static>,
+    ) -> Self {
+        RpcPlaceOrder {
+            top_extractor,
+            rpc_gateway,
+        }
+    }
+
     fn method_impl(
         &self,
         request: DirectRequest,
-    ) -> Result<(RpcInfo, bool, DirectRequestStatus), String> {
+    ) -> Result<(OrderUUID, bool, DirectRequestStatus), String> {
```

Comment:
Non-blocking call, so we can not return the OrderUUID here. Would a simple "order submit success" returned to the user suffice?